src/shared/utils/date-utils.ts

```diff
@@ -1,12 +1,45 @@
+export type Quarter = "Q1" | "Q2" | "Q3" | "Q4";
+
 const DELIMITER_DEFAULT = "-";
 const TIME_ZONE_DEFAULT = "Europe/London";
 
+// First month of each quarter, sorted by quarter number
+const QUARTER_FIRST_MONTHS: Record<Quarter, number> = {
+  Q1: 4,
+  Q2: 7,
+  Q3: 10,
+  Q4: 1,
+};
+
+// Quarter of each first month, sorted by month
+const QUARTERS_BY_FIRST_MONTH: Partial<Record<number, Quarter>> =
```

Comment:
How come `Partial`?

Reply:
yeah but from the type's perspective isn't `Partial<Record<number, Quarter>>` the same as `Record<number, Quarter>` since it's not a specific set of numbers?

Reply:
cool, thanks for the explanation :)